sample_consensus/include/pcl/sample_consensus/impl/lmeds.hpp

```diff
@@ -96,22 +97,36 @@ pcl::LeastMedianSquares<PointT>::computeModel (int debug_verbosity_level)
       ++skipped_count;
       continue;
     }
+    // Move all NaNs in distances to the end
+    const auto new_end = (sac_model_->getInputCloud()->is_dense ? distances.end() : std::partition (distances.begin(), distances.end(), [](double d){return !std::isnan (d);}));
+    const auto nr_valid_dists = std::distance (distances.begin (), new_end);
 
-    std::sort (distances.begin (), distances.end ());
     // d_cur_penalty = median (distances)
-    std::size_t mid = sac_model_->getIndices ()->size () / 2;
-    if (mid >= distances.size ())
+    const std::size_t mid = nr_valid_dists / 2;
+    PCL_DEBUG ("[pcl::LeastMedianSquares::computeModel] There are %lu valid distances remaining after removing NaN values.\n", nr_valid_dists);
+    if (nr_valid_dists == 0)
     {
       //iterations_++;
       ++skipped_count;
       continue;
     }
 
     // Do we have a "middle" point or should we "estimate" one ?
-    if (sac_model_->getIndices ()->size () % 2 == 0)
-      d_cur_penalty = (sqrt (distances[mid-1]) + sqrt (distances[mid])) / 2;
+    if ((nr_valid_dists % 2) == 0)
```

Comment:
`nr_valid_dists & 1` is better since modulo is a slow op and even clang trunk on O2 doesn't optimize it.

Reply:
I searched around and found [this answer](https://stackoverflow.com/a/160935). But tbh: it is done only once every iteration, so a few hundred times max. I do not see how it would have any effect on the performance.

Reply:
> I do not see how it would have any effect on the performance.

It doesn't, it's just something that's a reflex to me. asm difference is [visible](https://godbolt.org/z/KWE3rQ) under some conditions, but not always. So using bitwise and everywhere brings in uniformity. Like I said, both of these are minor comments.